crates/swift-bridge-ir/src/codegen/codegen_tests/transparent_enum_codegen_tests.rs

```diff
@@ -30,8 +30,8 @@ mod generates_enum_to_and_from_ffi_conversions_no_data {
             #[repr(C)]
             #[doc(hidden)]
             pub enum __swift_bridge__SomeEnum {
-                Variant1,
-                Variant2
+                Variant1(u8),
```

Comment:
We don't need the `u8` since `#[repr(C)]` enums are FFI safe.

Reply:
If Rust's `enum` type has no fields, the corresponding `union` and the corresponding `struct` in C is generated as something like this:

```rust
#[repr(C)]
pub enum MyEnum {
    A(i32),
    B,
}
```

```C
//C header 

typedef struct FieldsOfA {int32_t _0;} FieldsOfA;
typedef struct FieldsOfB {} FieldsOfB;  // This struct is empty.

union MyEnumFields {
    FieldsOfA A;
    FieldsOfB B;
};

typedef enum MyEnumTag { _A, _B, _C,} MyEnumTag;

typedef struct MyEnum {
    enum MyEnumTag tag;
    union MyEnumFields payload;
} MyEnum;

```


`FieldsOfB` is empty. Is this safety?

Reply:
Thanks for illustrating.

We don't need `FieldsOfB` in the `C` header union at all.

Look at `MyEnum::D` here https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#explicit-repr-annotation-with-c-compatibility

If you look at the corresponding C header file the `C` union doesn't have the `D` variant.

---

So, we can remove the `u8`, and not even generate the `FieldsOfB` at all.

For enums where where none of the fields are data-carrying we shouldn't generate the `union` and instead use the C header format

https://github.com/chinedufn/swift-bridge/blob/master/crates/swift-bridge-ir/src/codegen/codegen_tests/transparent_enum_codegen_tests.rs#L102-L103

---

Of course all of this is probably fairly minor since we'd only be saving a single byte of memory per enum... 
But I'd rather just do it right the first time.

Reply:
> If you look at the corresponding C header file the C union doesn't have the D variant.

I guess that we have to set some value to the `payload` variable. 

Considering the example above, do we set `FieldsOfA` to `palyload` when `tag` is `B`?

Something like: 
```Swift 
MyEnum(tag: B, payload:  MyEnumFields(A: FieldsOfA(_0: value)))
```

Reply:
As far as I understand, the `unit` type in Rust does not exist in the corresponding type in Swift. I checked [How Swift imports C APIs](https://github.com/apple/swift/blob/main/docs/HowSwiftImportsCAPIs.md#how-swift-imports-c-apis).

Reply:
It looks like we can use the default initializer. Explained here https://developer.apple.com/documentation/swift/using-imported-c-structs-and-unions-in-swift .

So, we want to generate something like this ` { var val = __swift_bridge__$EnumWithUnnamedData(); val.tag = __swift_bridge__$EnumWithUnnamedData$Variant3; return val } ()`

Here I've commented out the old return and added a new one that would work without specifying a payload:

```swift
public enum EnumWithUnnamedData {
    case Variant1(RustString, UInt32)
    case Variant2(Int32, UInt8)
    case Variant3
}
extension EnumWithUnnamedData {
    func intoFfiRepr() -> __swift_bridge__$EnumWithUnnamedData {
        switch self {
            case EnumWithUnnamedData.Variant1(let value_0, let value_1):
                return __swift_bridge__$EnumWithUnnamedData(tag: __swift_bridge__$EnumWithUnnamedData$Variant1, payload: __swift_bridge__$EnumWithUnnamedDataFields(Variant1: __swift_bridge__$EnumWithUnnamedData$FieldOfVariant1(_0: { let rustString = value_0.intoRustString(); rustString.isOwned = false; return rustString.ptr }(), _1: value_1)))
            case EnumWithUnnamedData.Variant2(let value_0, let value_1):
                return __swift_bridge__$EnumWithUnnamedData(tag: __swift_bridge__$EnumWithUnnamedData$Variant2, payload: __swift_bridge__$EnumWithUnnamedDataFields(Variant2: __swift_bridge__$EnumWithUnnamedData$FieldOfVariant2(_0: value_0, _1: value_1)))
            case EnumWithUnnamedData.Variant3:
//                return __swift_bridge__$EnumWithUnnamedData(tag: __swift_bridge__$EnumWithUnnamedData$Variant3, payload: __swift_bridge__$EnumWithUnnamedDataFields(Variant3: __swift_bridge__$EnumWithUnnamedData$FieldOfVariant3(_private: 123)))
            return { var val = __swift_bridge__$EnumWithUnnamedData(); val.tag = __swift_bridge__$EnumWithUnnamedData$Variant3; return val } ()
        }
    }
}
```

Reply:
Thanks you!!

Reply:
Yup! And then if none of the variants carry data the codegen should look like it did before this PR.

https://github.com/chinedufn/swift-bridge/blob/master/crates/swift-bridge-ir/src/codegen/codegen_tests/transparent_enum_codegen_tests.rs#L65-L81